### PR TITLE
 Replace assert by simpler but equivalent calls.

### PR DESCRIPTION
**Pull request by arturobernalg**

### CODE REVIEW

File: commons-numbers-angle/src/test/java/org/apache/commons/numbers/angle/AngleTest.java [hunk 1]
```diff
@@ -60,19 +60,19 @@ void testEquals() {
         Assertions.assertEquals(Angle.Turn.of(value), asTurn);
         Assertions.assertEquals(asTurn, asTurn);
         Assertions.assertNotEquals(asTurn, Angle.Turn.of(nextValue));
-        Assertions.assertFalse(asTurn.equals(null));
+        Assertions.assertNotEquals(null, asTurn);
```
Comment (aherbert) [inline commons-numbers-angle/src/test/java/org/apache/commons/numbers/angle/AngleTest.java:63]: Reverse argument order, null must be second.

File: commons-numbers-angle/src/test/java/org/apache/commons/numbers/angle/AngleTest.java [hunk 2]
```diff
@@ -60,19 +60,19 @@ void testEquals() {
...
 
         final Angle.Rad asRad = Angle.Rad.of(value);
         Assertions.assertEquals(Angle.Rad.of(value), asRad);
         Assertions.assertEquals(asRad, asRad);
         Assertions.assertNotEquals(asRad, Angle.Rad.of(nextValue));
-        Assertions.assertFalse(asRad.equals(null));
+        Assertions.assertNotEquals(null, asRad);
```
Comment (aherbert) [inline commons-numbers-angle/src/test/java/org/apache/commons/numbers/angle/AngleTest.java:69]: Reverse argument order, null must be second.

File: commons-numbers-angle/src/test/java/org/apache/commons/numbers/angle/AngleTest.java [hunk 3]
```diff
@@ -60,19 +60,19 @@ void testEquals() {
...
 
         final Angle.Deg asDeg = Angle.Deg.of(value);
         Assertions.assertEquals(Angle.Deg.of(value), asDeg);
         Assertions.assertEquals(asDeg, asDeg);
         Assertions.assertNotEquals(asDeg, Angle.Deg.of(nextValue));
-        Assertions.assertFalse(asDeg.equals(null));
+        Assertions.assertNotEquals(null, asDeg);
```
Comment (aherbert) [inline commons-numbers-angle/src/test/java/org/apache/commons/numbers/angle/AngleTest.java:75]: Reverse argument order, null must be second.

File: commons-numbers-complex/src/test/java/org/apache/commons/numbers/complex/ComplexTest.java [hunk 4]
```diff
@@ -1722,13 +1722,13 @@ void testNthRootInf() {
     @Test
     void testEqualsWithNull() {
         final Complex x = Complex.ofCartesian(3.0, 4.0);
-        Assertions.assertFalse(x.equals(null));
+        Assertions.assertNotEquals(null, x);
```
Comment (aherbert) [inline commons-numbers-complex/src/test/java/org/apache/commons/numbers/complex/ComplexTest.java:1725]: Reverse argument order, null must be second.

File: commons-numbers-field/src/test/java/org/apache/commons/numbers/field/FP64Test.java [hunk 5]
```diff
@@ -43,14 +43,14 @@ void testEquals() {
...
-        Assertions.assertTrue(a.equals(FP64.of(a.doubleValue())));
+        Assertions.assertEquals(a, FP64.of(a.doubleValue()));
         // Different value
-        Assertions.assertFalse(a.equals(b));
+        Assertions.assertNotEquals(a, b);
         // Different object
-        Assertions.assertFalse(a.equals(new Object()));
-        Assertions.assertFalse(a.equals(null));
+        Assertions.assertNotEquals(a, new Object());
+        Assertions.assertNotEquals(null, a);
```
Comment (aherbert) [inline commons-numbers-field/src/test/java/org/apache/commons/numbers/field/FP64Test.java:53]: Reverse argument order, null must be second.

File: commons-numbers-fraction/src/test/java/org/apache/commons/numbers/fraction/BigFractionTest.java [hunk 6]
```diff
@@ -625,10 +625,10 @@ void testSubtract() {
     @Test
     void testEqualsAndHashCode() {
         final BigFraction zero = BigFraction.of(0, 1);
-        Assertions.assertTrue(zero.equals(zero));
-        Assertions.assertFalse(zero.equals(null));
-        Assertions.assertFalse(zero.equals(new Object()));
-        Assertions.assertFalse(zero.equals(Double.valueOf(0)));
+        Assertions.assertEquals(zero, zero);
+        Assertions.assertNotEquals(null, zero);
```
Comment (aherbert) [inline commons-numbers-fraction/src/test/java/org/apache/commons/numbers/fraction/BigFractionTest.java:629]: Reverse argument order, null must be second.

File: commons-numbers-quaternion/src/test/java/org/apache/commons/numbers/quaternion/QuaternionTest.java [hunk 7]
```diff
@@ -378,7 +378,7 @@ final void testObjectEquals() {
         final Quaternion q3 = Quaternion.of(one, Math.nextUp(one), one, one);
         Assertions.assertNotEquals(q3, q1);
 
-        Assertions.assertFalse(q3.equals("bar"));
+        Assertions.assertNotEquals("bar", q3);
```
Comment (aherbert) [inline commons-numbers-quaternion/src/test/java/org/apache/commons/numbers/quaternion/QuaternionTest.java:381]: Reverse argument order

File: commons-numbers-fraction/src/test/java/org/apache/commons/numbers/fraction/FractionTest.java [hunk 8]
```diff
@@ -463,10 +463,10 @@ void testSubtract() {
     @Test
     void testEqualsAndHashCode() {
         final Fraction zero = Fraction.of(0, 1);
-        Assertions.assertTrue(zero.equals(zero));
-        Assertions.assertFalse(zero.equals(null));
-        Assertions.assertFalse(zero.equals(new Object()));
-        Assertions.assertFalse(zero.equals(Double.valueOf(0)));
+        Assertions.assertEquals(zero, zero);
+        Assertions.assertNotEquals(null, zero);
```
Comment (aherbert) [inline commons-numbers-fraction/src/test/java/org/apache/commons/numbers/fraction/FractionTest.java:467]: Reverse argument order, null must be second.

File: commons-numbers-angle/src/test/java/org/apache/commons/numbers/angle/AngleTest.java [hunk 9]
```diff
@@ -60,19 +60,19 @@ void testEquals() {
...
 
         final Angle.Rad asRad = Angle.Rad.of(value);
         Assertions.assertEquals(Angle.Rad.of(value), asRad);
         Assertions.assertEquals(asRad, asRad);
         Assertions.assertNotEquals(asRad, Angle.Rad.of(nextValue));
-        Assertions.assertFalse(asRad.equals(null));
+        Assertions.assertNotEquals(null, asRad);
 
         final Angle.Deg asDeg = Angle.Deg.of(value);
         Assertions.assertEquals(Angle.Deg.of(value), asDeg);
```
Comment (aherbert) [inline commons-numbers-angle/src/test/java/org/apache/commons/numbers/angle/AngleTest.java:72]: Reverse these arguments for consistency

File: commons-numbers-angle/src/test/java/org/apache/commons/numbers/angle/AngleTest.java [hunk 10]
```diff
@@ -60,19 +60,19 @@ void testEquals() {
         Assertions.assertEquals(Angle.Turn.of(value), asTurn);
         Assertions.assertEquals(asTurn, asTurn);
         Assertions.assertNotEquals(asTurn, Angle.Turn.of(nextValue));
-        Assertions.assertFalse(asTurn.equals(null));
+        Assertions.assertNotEquals(null, asTurn);
 
         final Angle.Rad asRad = Angle.Rad.of(value);
         Assertions.assertEquals(Angle.Rad.of(value), asRad);
```
Comment (aherbert) [inline commons-numbers-angle/src/test/java/org/apache/commons/numbers/angle/AngleTest.java:66]: Reverse these arguments for consistency

File: commons-numbers-angle/src/test/java/org/apache/commons/numbers/angle/AngleTest.java [hunk 11]
```diff
@@ -60,19 +60,19 @@ void testEquals() {
         Assertions.assertEquals(Angle.Turn.of(value), asTurn);
```
Comment (aherbert) [inline commons-numbers-angle/src/test/java/org/apache/commons/numbers/angle/AngleTest.java:60]: Reverse these arguments for consistency

**Review by aherbert — CHANGES_REQUESTED** [review]
When using `assertNotEquals` the `null` argument must be second so that the assertion calls the equals method on the class being tested with the `null` as the argument.

In general the class where the equals method is being tested should always be first.

File: commons-numbers-fraction/src/test/java/org/apache/commons/numbers/fraction/FractionTest.java [hunk 12]
```diff
@@ -463,10 +463,10 @@ void testSubtract() {
...
+        Assertions.assertNotEquals(zero, new Object());
+        Assertions.assertNotEquals(zero, Double.valueOf(0), 0.0);
```
Comment (aherbert) [inline commons-numbers-fraction/src/test/java/org/apache/commons/numbers/fraction/FractionTest.java:469]: This is creating a compile failure in the CI build. Remove the `0.0` tolerance argument. This is a test for Object.equals and not floating-point equivalence of doubles.

**Comment by arturobernalg** [comment]
> When using `assertNotEquals` the `null` argument must be second so that the assertion calls the equals method on the class being tested with the `null` as the argument.
> 
> In general the class where the equals method is being tested should always be first.

hi @aherbert 
Changed
TY

File: commons-numbers-fraction/src/test/java/org/apache/commons/numbers/fraction/FractionTest.java [hunk 13]
```diff
@@ -463,10 +463,10 @@ void testSubtract() {
...
     void testEqualsAndHashCode() {
         final Fraction zero = Fraction.of(0, 1);
-        Assertions.assertTrue(zero.equals(zero));
-        Assertions.assertFalse(zero.equals(null));
-        Assertions.assertFalse(zero.equals(new Object()));
-        Assertions.assertFalse(zero.equals(Double.valueOf(0)));
+        Assertions.assertEquals(zero, zero);
+        Assertions.assertNotEquals(zero, null);
+        Assertions.assertNotEquals(zero, new Object());
+        Assertions.assertNotEquals(zero, (double) 0);
```
Comment (aherbert) [inline commons-numbers-fraction/src/test/java/org/apache/commons/numbers/fraction/FractionTest.java:469]: Do not cast 0 to `double`. This requires an object. Use `Double.valueOf(0)`.

File: commons-numbers-fraction/src/test/java/org/apache/commons/numbers/fraction/BigFractionTest.java [hunk 14]
```diff
@@ -625,10 +625,10 @@ void testSubtract() {
...
+        Assertions.assertNotEquals(zero, new Object());
+        Assertions.assertNotEquals(zero, Double.valueOf(0), 0.0);
```
Comment (aherbert) [inline commons-numbers-fraction/src/test/java/org/apache/commons/numbers/fraction/BigFractionTest.java:631]: Remove the delta tolerance here. And two lines above please reverse the order of null and zero.

File: commons-numbers-fraction/src/test/java/org/apache/commons/numbers/fraction/BigFractionTest.java (hunk 6 above)
Comment (aherbert) [reply commons-numbers-fraction/src/test/java/org/apache/commons/numbers/fraction/BigFractionTest.java:629]: Ping. This once still needs to be reversed.

**Pull request merged**